src/wesnoth.cpp

```diff
@@ -898,6 +898,8 @@ static int do_gameloop(commandline_options& cmdline_opts)
 		case gui2::dialogs::title_screen::RELOAD_GAME_DATA:
 			gui2::dialogs::loading_screen::display([&config_manager]() {
 				config_manager.reload_changed_game_config();
+				gui2::init();
```

Comment:
I suppose this happens on F5 and addon install? In the addon install case it doesn't really make sense to parse mainline themes again. Don't know if you can currently decide which case it is though and it might not make much of a difference..

Reply:
> I suppose this happens on F5 and addon install
Yes.